docs for multi domain (#3409)
RasaHQ/rasa: docs/nlu/multi-skill-assistants.rst

:desc: Iterate quickly by developing reusable building blocks of AI assistant skills
       and combining them at training time.

.. _multi_skill_bots:

Multi Skill Assistants
======================

Rasa supports building AI assistants from multiple contextual AI assistant projects.

> This allows it to develop reusable building blocks of skills which you can reuse among

Reviewer: ```suggestion
This allows for the development of reusable building blocks of skills which you can use within
```

> your projects. One skill could for example handle chitchat, while the other skill

Reviewer: ```suggestion
your different projects. For example, one skill could handle chitchat while another skill
```

> is responsible for greeting your users. You can develop each skill isolated and then

Reviewer: ```suggestion
is responsible for greeting your users. You can develop skills in isolation, and then
```

> import them for the combined training of contextual AI assistant.

Reviewer: ```suggestion
import them for the combined training of a contextual AI assistant.
```

An example directory structure could e.g. look like this:

.. code-block:: bash

    .
    ├── config.yml

>     └── data

Reviewer: I would think it would make more sense for this dir to be called something like `projects` since each of them already has it's own `data`, and would reflect `building AI assistants from multiple contextual AI assistant projects`

Reviewer: or `skills`. I think we should choose that we are either importing skills or projects, and then use that language overall (e.g. below we talk about importing skills, but above talk about building from projects)

        ├── GreetBot
        │   ├── data
        │   │   ├── nlu.md
        │   │   └── stories.md
        │   └── domain.yml
        └── MoodBot

>             ├── config.yml

Reviewer: Should MoodBot have its own config? If yes, shouldn't GreetBot have one too?

Reviewer: I see that maybe the only thing in moodbot's config is importing greetbot -- if this is the case then we should make it clear that nothing else (e.g. policies, pipeline) goes in the `config.yml` for the lower projects

Author: I agree - will add this

            ├── data
            │   ├── nlu.md
            │   └── stories.md
            └── domain.yml

In this example the contextual AI assistant imports the ``MoodBot`` skill which in turn
imports the ``GreetBot`` skill. Skill imports are defined in the configuration files of
each project. The ``config.yml`` in the root project e.g. looks like this:

.. code-block:: yaml

    imports:

>     - data/MoodBot

Reviewer: same as above in L22

The configuration file of the ``MoodBot`` in turn references the ``GreetBot``:

.. code-block:: yaml

    imports:
    - ../GreetBot

> Rasa uses relatives paths from the referencing configuration files to import skills.

Reviewer: ```suggestion
Rasa uses relative paths from the referencing configuration files to import skills.
```
